.github/workflows/ci.yaml

```diff
@@ -0,0 +1,240 @@
+name: GitHub Actions CI
+
+# Cache is used for all Windows and macOS dependencies (size approx. 1230 * 2 + 1850 = 4310 MiB)
+# Cache is not used for Ubuntu builds, because it already has all dependencies except
+# the appropriate libtorrent version, which only takes 3-5 minutes to build from source anyway
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ master ]
+
+env:
+  # Qt: 5.15.0
+  # libtorrent: RC_1_2 HEAD, 1.2.10
+  VCPKG_COMMIT: 32eccc18191fbb57b159784a1724d2d00613ae82
+  VCPKG_DEST_MACOS: /Users/runner/qbt_tools/vcpkg
+  VCPKG_DEST_WIN: C:\qbt_tools\vcpkg
+  UBUNTU_LIBTORRENT_VERSION: libtorrent-1.2.10
+
+# NOTE: for client code built with >= C++14, libtorrent 1.2.x must be built with >= C++14 as well.
+# Compilers that don't default to >= C++14 must be overriden.
+# See https://github.com/arvidn/libtorrent/issues/4965.
```

Comment:
@Chocobo1 I disagree, it's a useful note for maintainers, and demystifies the purpose of the `force AppleClang to compile libtorrent with C++14` step. Once we drop libtorrent 1.2 support, and assuming libtorrent >= 2.x won't have similar ABI issues (check the linked issue report), we can remove such comments.

Reply:
>I disagree, it's a useful note for maintainers

Well, I would argue if the maintainer doesn't know this, then he isn't really suitable for the job.
I think this issue is widely known among the developers working on this project.

Reply:
@Chocobo1 

Why make it harder for people to discover these kinds of things? Prospecting contributors that are confused by the code that these comments refer to will have a much harder time making sense of it until they find on their own the information that is present in these comments currently. Imagine being new to the project and wondering why the C++ standard is forced in the macOS CI. Without these comments, you'll waste much more time finding out the reason.

Plus, will you remember this issue easily without such comments 3-4 months for now? I won't. We all have more important stuff to do and remember. Without such comments, it's much more effort to dig out the relevant details and reasons behind related changes in the future if we need to refer back to this issue.

Reply:
>Why make it harder for people to discover these kinds of things? 

Those aren't special cases when working with C++ codebases. It is universally known that the dependencies *should* be compiled with the same compiler & C++ version mode, otherwise weird things ensues.
At least you should move this comment near to the respective command and make it succinct that will fit in a single line.

Reply:
@Chocobo1 

The thing is, everything should also work fine if libtorrent is compiled with a lower C++ version than qBittorrent, and libtorrent's officially stated minimum C++ standard requirement is 11. A lot of people, including many official and community distro packages have been building qBittorrent against C++11-built libtorrent.

Recently I had to deal with a torrent (heh) of issues from several users and official/community distro packages that were working but suddenly broke because they were neither building libtorrent with C++14 _or_ C++11 and defining `TORRENT_CXX11ABI`.

It is very easy to accidentally run into this issue, and it is not easy to figure out the root of the problem just from the ensuing error message at buildtime.

Plus, while currently only the macOS build requires a workaround, the issue is universal, and it might be the case that in the future we may also need to apply the same workaround to other systems. That's why the comment is at the top.

The current explanation is only 2 more lines, and it's already incomplete (it does not mention the existence of a workaround when building libtorrent with C++11).

Reply:
>Plus, while currently only the macOS build requires the workaround of forcing the compiler version in practice, the issue is universal and OS-agnostic in principle, and it might be the case that in the future we may also need to apply the same workaround to other systems. That's why the comment is at the top.

Your whole reasoning made me think CI script is (still) not the right place you want to put the info at.

Reply:
@Chocobo1 

Where do you suggest I put it instead?

Reply:
>Where do you suggest I put it instead?

Where someone looks for info for compiling qbt, I suppose this would be the guide in our wiki.

Reply:
@Chocobo1 

> The CI script should mainly contain CI actions and comments about them, not detailed descriptions to some particular issue.

This 3-line comment is not a "detailed description about an issue". I'd argue it falls precisely into the category of "comments about CI actions". Does it not?

Maybe I can change the wording of the second line to make it really obvious that it has bearing on other things written in the file:

"\# **Currently, in all CI jobs,** ~C~compilers that don't default to >= C++14 ~must be~ **are** overridden **to >= C++14 mode to avoid the issue.**"

Seriously what do you have against little snippets of documentation in the code?

Reply:
>Seriously what do you have against little snippets of documentation in the code, to explain non-obvious decisions?

https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491151694

Reply:
@Chocobo1 

Extracted from https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491725039, for visibility:

>OK, I already explained why I think it makes more sense for this to be at the top, but I'm willing to compromise here. How about:
>
>```
># NOTE: Avoids a libtorrent ABI issue. See https://github.com/arvidn/libtorrent/issues/4965
>```
>
>on top of the `Force AppleClang ...` step? 1 line, doesn't extend past column 100 even when accounting for indentation.